apps/server/src/modules/authorization-rules/rules/instance.rule.spec.ts

```diff
@@ -77,123 +72,294 @@ describe(InstanceRule.name, () => {
 
-					expect(result).toEqual(false);
+					expect(result).toBe(false);
 				});
 			});
 
-			describe('when the user has read permission', () => {
-				const setup = () => {
-					const user = userFactory.asStudent().build();
-					const instance = instanceFactory.build();
-					const context: AuthorizationContext = {
-						action: Action.read,
-						requiredPermissions: [Permission.FILESTORAGE_VIEW],
-					};
+			describe('when user has CAN_EXECUTE_INSTANCE_OPERATIONS and INSTANCE_EDIT', () => {
+				const setup = () =>
+					createParamsFromUserAndContextPermissions([
+						Permission.INSTANCE_EDIT,
+						Permission.CAN_EXECUTE_INSTANCE_OPERATIONS,
+					]);
 
-					return {
-						user,
-						instance,
-						context,
-					};
-				};
+				it('should be return true', () => {
+					const { user, instance, context } = setup();
+
+					const result = rule.hasPermission(user, instance, context);
+
+					expect(result).toBe(true);
+				});
+			});
+
+			describe('when user has INSTANCE_VIEW', () => {
+				const setup = () => createParamsFromUserAndContextPermissions([Permission.INSTANCE_VIEW]);
+
+				it('should be return false, because it is not implemented', () => {
+					const { user, instance, context } = setup();
+
+					const result = rule.hasPermission(user, instance, context);
+
+					expect(result).toBe(false);
+				});
+			});
+
+			describe('when user has INSTANCE_EDIT', () => {
+				const setup = () => createParamsFromUserAndContextPermissions([Permission.INSTANCE_EDIT]);
+
+				it('should be return false, because it is not implemented', () => {
+					const { user, instance, context } = setup();
+
+					const result = rule.hasPermission(user, instance, context);
+
+					expect(result).toBe(false);
+				});
+			});
+
+			describe('when user has no required permission', () => {
+				const setup = () => createParamsFromUserAndContextPermissions([]);
+
+				it('should be return false', () => {
+					const { user, instance, context } = setup();
+
+					const result = rule.hasPermission(user, instance, context);
+
+					expect(result).toBe(false);
+				});
+			});
+
+			describe('when additional required permission is passed and user with CAN_EXECUTE_INSTANCE_OPERATIONS has it.', () => {
+				const setup = () =>
+					createParamsFromUserAndContextPermissions(
+						[Permission.INSTANCE_EDIT, Permission.CAN_EXECUTE_INSTANCE_OPERATIONS, Permission.TEST_ONLY],
+						[Permission.TEST_ONLY]
+					);
+
+				it('should be return true', () => {
+					const { user, instance, context } = setup();
+
+					const result = rule.hasPermission(user, instance, context);
+
+					expect(result).toBe(true);
+				});
+			});
+
+			describe('when additional required permission is passed but user with CAN_EXECUTE_INSTANCE_OPERATIONS do not have it.', () => {
+				const setup = () =>
+					createParamsFromUserAndContextPermissions(
+						[Permission.INSTANCE_EDIT, Permission.CAN_EXECUTE_INSTANCE_OPERATIONS],
+						[Permission.TEST_ONLY]
```

Comment:
Änderungen in anderen Bereichen bzg. Permissions haben durch TEST_ONLY keine Auswirkungen auf diese Tests.